### PR TITLE
Int 1109 repository groovy wrapper

### PR DESCRIPTION
**Pull request by fernau**
# Description
Create groovy wrapper around nexus-platform-sdk

### CODE REVIEW

File: cli/src/main/groovy/NexusPublisher.groovy [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import com.sonatype.nexus.api.common.Authentication
+import com.sonatype.nexus.api.common.ServerConfig
+import com.sonatype.nexus.api.repository.v3.DefaultAsset
+import com.sonatype.nexus.api.repository.v3.DefaultComponent
+import com.sonatype.nexus.api.repository.v3.RepositoryManagerV3ClientBuilder
+
+import groovy.cli.commons.CliBuilder
+
+/**
+ * Copyright (c) 2016-present Sonatype, Inc. All rights reserved.
```
Comment (bigspotteddog) [inline cli/src/main/groovy/NexusPublisher.groovy]: Should this date be 2016? Or, more recent.

File: cli/src/main/groovy/NexusPublisher.groovy [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+cli = new CliBuilder(usage: 'Repository', expandArgumentFiles: true)
+cli.h(type: Boolean, longOpt: 'help', 'Prints this help text')
+cli._(longOpt: 'serverurl', 'URL of nexus repository manager server', convert: {URI.create(it)}, required: true)
+cli.u(type: String, longOpt: 'username', 'Username', required: true)
+cli.p(type: String, longOpt: 'password', 'Password', required: true)
+cli.f(type: String, longOpt: 'format', 'Artifact format. Examples: maven2', required: true)
+cli._(longOpt: 'filename', 'Filename to upload', convert: {new File(it)}, required: true)
+cli.C(args:2, valueSeparator:'=', argName:'key=value', 'Component coordinates, can be used multiple times. Example: -CgroupId=com.example -CartifactId=myapp -Cversion=1.0', required: true)
+cli.A(args:2, valueSeparator:'=', argName:'key=value', 'Asset attributes, can be used multiple times. Example: -Aextension=jar -Aclassifier=bin', required: true)
+cli.r(type: String, longOpt: 'repository', 'Name of target repository on Nexus. Example: maven-releases', required: true)
```
Comment (rjackson64840) [inline cli/src/main/groovy/NexusPublisher.groovy:36]: Can you wrap the above lines so they all fit in the review?

**Review by bigspotteddog — APPROVED** [review]
+1 with one question

File: cli/src/main/groovy/NexusPublisher.groovy [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+
+@Grab(group='org.slf4j', module='slf4j-simple', version='1.7.25')
+@Grab(group='com.sonatype.nexus', module='nexus-platform-api', version='3.3.20181025-095610.1ce86ad')
+
+cli = new CliBuilder(usage: 'Repository', expandArgumentFiles: true)
```
Comment (rjackson64840) [inline cli/src/main/groovy/NexusPublisher.groovy:25]: After studying this for a bit it looks like these are all definitions of command line arguments that we want to process and that this happens with the 'cli.parse' command below.  My perception before I started this review was that we were producing and executing a command line, but the opposite appears to be true.  This is essentially like a java main() and we're consuming the command line and then doing some processing on that.

I wonder if an overview sentence or two would be helpful in this case.

**Review by rjackson64840 — APPROVED** [review]
LGTM as best I can tell.  Maybe just a couple comments to help shorten the learning curve.

File: cli/src/main/groovy/NexusPublisher.groovy (hunk 1 above)
Comment (fernau) [reply cli/src/main/groovy/NexusPublisher.groovy]: fixed. thought that would happen automatically after updating headers.txt ;(

File: cli/src/main/groovy/NexusPublisher.groovy (hunk 2 above)
Comment (fernau) [reply cli/src/main/groovy/NexusPublisher.groovy:36]: wrapped

**Pull request merged**